File: Java/Hello.java
import java.util.Scanner;

Class Main {
    public static void main(String args[]) {
        Scanner scanner = new Scanner(System.in);
        String buf = new String();
        System.out.println("please input your name:");
        buf = scanner.nextLine();
        System.out.println("Hello, " + buf + "!\n");
    }
}

File: CPP/Hello.cpp
#include <iostream>

const int NAME_LEN = 30;

int main()
{
    char *name = new char[NAME_LEN];
    std::cout << "please inputs your name" << std::endl;
    std::cin.getline(name, NAME_LEN);
    
    std::cout << std::endl << "Hello, " << *name << "!" << std::endl;
    
    delete[] name;
    return 0;
}